### ISSUE TITLE
Make all characters invisible

### ISSUE THREAD

**Issue by ariadesu** [fr33mind/Belle#18]
It would speed things up either for narrator type characters, or for showing a CG fullscreen. Preferably with it remembering the state of who was visible before the toggle so you can bring everybody back with a single click.

**Comment by fr33mind** [comment]
I see what you mean. Yes, sounds like a good idea. I could add a "hide all" and "show all" actions, which you could configure to work with an object type (i.e. character, button, etc) or by name or all. Sounds good?

**Comment by ariadesu** [comment]
Yes. I would put them under the current hide and show. Instead of hiding individual, you can choose "toggle all visible" or similar in the properties for the hide event. This way the possibility space seems smaller, making it less intimidating to get started. 

But managing sprites by individual hide and show commands in general isn't great, because it's not very visual. I recommend a timeline next to the actions list that controls when each object is visible. Mockup to better explain what I mean. https://i.imgur.com/HGO5VMT.png

**Comment by fr33mind** [comment]
As for the first part, i agree, it's better to put it in "hide" and "show" instead of creating new actions, thanks.

As for the second part, those green bars would exist one per object/character and would mean the object/character was visible during those actions, right?
If that's the case i can understand it would be easier to have that visual aid. However i see some problems that makes it difficult to implement that now:

Firstly, currently, the editor has no idea how the game will run (that's the job of the engine). It only shows a static representation of the game, so doing that could give a misleading visual representation, if, for example, you had GoToLabel or Branch actions somewhere in the middle. You see what i mean?
Secondly, even if the editor knew exactly how the game would run, i'm not sure how that visual representation would work, because there are always choices that are taken by the player at runtime and that makes it difficult to represent in a clean way in the editor.

So with that said, it seems like a nice feature, don't get me wrong. However it also seems that it would take lots of time to develop in order to create a reliable representation. Not only that, but more importantly for something that it's not that vital to the development.

**Comment by ariadesu** [comment]
Is the issue that the editor does not reliably know what objects are present in a scene? Or do you mean what state each object is in?

The way I see it, if the actions list is cleaned up by moving visibility to a second column, I don't see why every "tick" shouldn't reinforce the state of the game. The bars in the timeline should not equate to a starting show and ending hide action, but instead a visibility check per tick. Of course, you want to be able to influence this at runtime. Regardless, you do not want the visibility to be incidental. Maintaining consistency after a transition is less important than supporting the author's intent. The author should be given the tools to maintain consistency, but the program should want to show the scene as closely to how it was authored as possible. 

If you implement a global visibility timeline that can be overruled at runtime with hide and show commands, the UX might become slightly frustrating as you're not sure what is influencing visibility. However, it does also make the choice much more deliberate by the user. 

The upside to the current method, is that it's very clear when something is visible or not. Because it's set once. However the downside is that it's also very easy to lose track.

**Comment by fr33mind** [comment]
Basically the editor doesn't track the changes of the actions. For example, it just knows that action A has an effect on object B, but it doesn't know what has happened before arriving at action A. Do you understand?

I'm not sure i understand exactly what you want. Are the green bars customizable by the user in order to define where the objects should appear or not or, on the other hand, are just a reflection of the actions' execution?

**Comment by ariadesu** [comment]
I think I fully understand how it works. 

The green bars are user set and intended to replace "hide" and "show" actions. For every step down the "actions" timeline, you would specify a hide and a show for every object in the scene. In other words, every non-visibility action is preceded by looping over every object in the scene and either hiding or showing it, depending on what the user has chosen. 

Sometimes you want to overwrite this depending on player choices, but I think authors learning two methods of interacting with hiding and showing is worth the tradeoff, because it's such a big usability improvement.

**Comment by fr33mind** [comment]
Ok, i understand. That way makes more sense. However i think the implementation needs to be a bit more thought out. I'm not sure having green bars alongside the actions is a very good idea.

In any case, still this is not a vital feature and therefore not high priority, so i'll have to leave this request for a later version.
